File: example/src/main/java/com/example/bundlerexample/MainActivity.java
package com.example.bundlerexample;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.example.bundlerexample.model.User;
import com.example.bundlerexample.test.C;
import com.hendraanggrian.bundler.Bundler;

import butterknife.BindView;

public class MainActivity extends BaseActivity implements View.OnClickListener {

    @BindView(R.id.toolbar_main) Toolbar toolbar;
    @BindView(R.id.button_main_example1) Button buttonExample1;
    @BindView(R.id.button_main_example2) Button buttonExample2;

    @Override
    protected int getContentLayout() {
        return R.layout.activity_main;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setSupportActionBar(toolbar);
        buttonExample1.setOnClickListener(this);
        buttonExample2.setOnClickListener(this);

        if (true) {
            Log.d("ASD", Bundler.wrap(C.class, "Hello", "World", "!").toString());
        }
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.button_main_example1:
                startActivity(new Intent(this, Example1Activity.class)
                        .putExtras(Bundler.wrap(Example1Activity.class, true, false)));
                break;
            case R.id.button_main_example2:
                User user = new User();
                user.name = "Hendra";
                startActivity(new Intent(this, Example2Activity.class)
                        //.putExtra(E.Example2Activity.user, Parcels.wrap(user)));
                        .putExtras(Bundler.wrap(Example2Activity.class, user)));
                break;
        }
    }
}

File: example/src/main/java/com/example/bundlerexample/MainActivity.kt
package com.example.bundlerexample

import android.animation.Animator
import android.animation.AnimatorListenerAdapter
import android.animation.AnimatorSet
import android.content.res.Resources
import android.os.Bundle
import android.support.v4.view.animation.FastOutSlowInInterpolator
import android.support.v7.widget.CardView
import android.support.v7.widget.Toolbar
import android.view.View
import android.widget.Button
import butterknife.BindView
import com.hendraanggrian.bundler.BindState
import com.hendraanggrian.bundler.Bundler
import com.hendraanggrian.reveallayout.RevealableLayout

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
class MainActivity(override val contentLayout: Int = R.layout.activity_main) : BaseActivity() {

    companion object {
        fun createAnimatorSet(resources: Resources, animators: Collection<Animator>): AnimatorSet {
            val set = AnimatorSet()
            set.playTogether(animators)
            set.interpolator = FastOutSlowInInterpolator()
            set.duration = resources.getInteger(android.R.integer.config_longAnimTime).toLong()
            return set
        }
    }

    @BindState @JvmField var cardViewShown: Boolean = false

    @BindView(R.id.revealableLayout) lateinit var revealableLayout: RevealableLayout
    @BindView(R.id.button) lateinit var button: Button
    @BindView(R.id.cardView) lateinit var cardView: CardView
    @BindView(R.id.toolbar) lateinit var toolbar: Toolbar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        button.setOnClickListener {
            val set = createAnimatorSet(resources, revealableLayout.animateTo(button, cardView))
            set.addListener(object : AnimatorListenerAdapter() {
                override fun onAnimationStart(animation: Animator?) {
                    button.visibility = View.INVISIBLE
                    cardView.visibility = View.VISIBLE
                }

                override fun onAnimationEnd(animation: Animator?) {
                    cardViewShown = true
                }
            })
            set.start()
        }
        toolbar.setNavigationOnClickListener {
            val set = createAnimatorSet(resources, revealableLayout.animateTo(button, cardView, true))
            set.addListener(object : AnimatorListenerAdapter() {
                override fun onAnimationEnd(animation: Animator?) {
                    button.visibility = View.VISIBLE
                    cardView.visibility = View.INVISIBLE
                    cardViewShown = false
                }
            })
            set.start()
        }

        savedInstanceState?.let {
            Bundler.bindStates(this, it)
            if (cardViewShown) {
                button.visibility = View.INVISIBLE
                cardView.visibility = View.VISIBLE
            }
        }
    }

    override fun onSaveInstanceState(outState: Bundle?) {
        super.onSaveInstanceState(outState)
        Bundler.onSaveInstanceState(outState!!, javaClass, cardViewShown)
    }
}